hs_access_control/models/group.py

```diff
@@ -162,44 +300,35 @@ def get_resources_with_explicit_access(self, this_privilege):
         # b) There is no lower privilege in either group privileges for the object.
         # c) Thus X is the effective privilege of the object.
         if this_privilege == PrivilegeCodes.OWNER:
-            return BaseResource.objects.filter(r2grp__privilege=this_privilege,
-                                               r2grp__group=self.group)\
-                                       .exclude(pk__in=BaseResource.objects
-                                                .filter(r2grp__group=self.group,
-                                                        r2grp__privilege__lt=this_privilege))
+            return BaseResource.objects.none()  # groups cannot own resources
 
         elif this_privilege == PrivilegeCodes.CHANGE:
             # CHANGE does not include immutable resources
             return BaseResource.objects.filter(raccess__immutable=False,
                                                r2grp__privilege=this_privilege,
-                                               r2grp__group=self.group)\
-                                       .exclude(pk__in=BaseResource.objects
-                                                .filter(r2grp__group=self.group,
-                                                        r2grp__privilege__lt=this_privilege))
+                                               r2grp__group=self.group)
+            # there are no excluded resources; maximum privilege is CHANGE
 
-        else:  # this_privilege == PrivilegeCodes.ViEW
+        else:  # this_privilege == PrivilegeCodes.VIEW
             # VIEW includes CHANGE & immutable as well as explicit VIEW
-            view = BaseResource.objects.filter(r2grp__privilege=this_privilege,
-                                               r2grp__group=self.group)\
-                .exclude(pk__in=BaseResource.objects.filter(r2grp__group=self.group,
-                                                            r2grp__privilege__lt=this_privilege))
-
-            immutable = BaseResource.objects.filter(raccess__immutable=True,
-                                                    r2grp__privilege=PrivilegeCodes.CHANGE,
-                                                    r2grp__group=self.group)\
-                                            .exclude(pk__in=BaseResource.objects.filter(
-                                                raccess__immutable=True,
-                                                r2grp__group=self.group,
-                                                r2grp__privilege__lt=PrivilegeCodes.CHANGE))
-
-            return BaseResource.objects.filter(Q(pk__in=view) | Q(pk__in=immutable)).distinct()
+            excluded = BaseResource.objects.filter(raccess__immutable=False,
```

Comment:
Do we need this `excluded`? Seems like the resources that are part of the `excluded` are not going to be part of the main query (outer query) results anyway?

Reply:
Correct. Fixed.